openverse_catalog/dags/providers/factory_utils.py

```diff
@@ -0,0 +1,137 @@
+import inspect
+import logging
+import time
+from types import FunctionType
+from typing import Callable, Sequence
+
+from airflow.models import TaskInstance
+from common.constants import MediaType
+from common.storage.media import MediaStore
+
+
+logger = logging.getLogger(__name__)
+
+
+def _load_provider_script_stores(
+    ingestion_callable: Callable,
+    media_types: list[MediaType],
+    **kwargs,
+) -> dict[str, MediaStore]:
+    """
+    Load the stores associated with a provided ingestion callable. This callable is
+    assumed to be a legacy provider script and NOT a provider data ingestion class.
+    """
+    # Stores exist at the module level, so in order to retrieve the output values we
+    # must first pull the stores from the module.
+    module = inspect.getmodule(ingestion_callable)
+    stores = {}
+    for media_type in media_types:
+        store = getattr(module, f"{media_type}_store", None)
+        if not store:
+            continue
+        stores[media_type] = store
+
+    if len(stores) != len(media_types):
+        raise ValueError(
+            f"Expected stores in {module.__name__} were missing: "
+            f"{list(set(media_types) - set(stores))}"
+        )
+    return stores
+
+
+def generate_tsv_filenames(
+    ingestion_callable: Callable,
+    media_types: list[MediaType],
+    ti: TaskInstance,
+    args: Sequence = None,
+) -> None:
+    """
+    Calculate the output directories for each media store to XComs. Output locations
+    are pushed under keys with the format `<media-type>_tsv`. This is a temporary
+    workaround due to the nature of the current provider scripts. Once
+    https://github.com/WordPress/openverse-catalog/issues/229 is addressed and the
+    provider scripts are refactored into classes, we can alter the process by which
+    MediaStores create/set their output directory so that they are always expected
+    to receive it.
+    """
+    args = args or []
+    logger.info("Pushing available store paths to XComs")
+
+    # TODO: This entire branch can be removed when all of the provider scripts have been
+    # TODO: refactored to subclass ProviderDataIngester.
+    if isinstance(ingestion_callable, FunctionType):
+        stores = _load_provider_script_stores(ingestion_callable, media_types)
+
+    else:
+        # A ProviderDataIngester class was passed instead. First we initialize the
+        # class, which will initialize the media stores and DelayedRequester.
+        logger.info(
+            f"Initializing ProviderIngester {ingestion_callable.__name__} in"
+            f"order to generate store filenames."
+        )
+        ingester = ingestion_callable(*args)
+        stores = ingester.media_stores
```

Comment:
You reference refactoring this down the line in the PR description so this is probably already the plan, but -- I'd love to pull the filename generation out of the media store and actually perform the logic here instead of initializing the stores twice. This would make it easier to add conditional logic (in my case, like adding a `date` suffix for tsvs generated during reingestion) without having to make the MediaStore aware of those parameters.

Totally reasonable to wait until the provider scripts are refactored though, to avoid duplicating the logic. 👍

Reply:
100% agree! And yea, I'd rather have the paths look similar now while we still have two, and then totally shift stuff around when we only have the one to worry about.